plugin/trino-hudi/src/main/java/io/trino/plugin/hudi/query/HudiReadOptimizedFileLister.java

```diff
@@ -0,0 +1,232 @@
+            HudiTableHandle tableHandle,
+            HoodieTableMetaClient metaClient,
+            HiveMetastore hiveMetastore,
+            Table hiveTable,
+            List<HiveColumnHandle> partitionColumnHandles,
+            boolean shouldSkipMetastoreForPartition)
+    {
+        this.metadataConfig = metadataConfig;
+        this.engineContext = engineContext;
+        this.tableHandle = tableHandle;
+        this.tableName = tableHandle.getSchemaTableName();
+        this.metaClient = metaClient;
+        this.hiveMetastore = hiveMetastore;
+        this.hiveTable = hiveTable;
+        this.partitionColumnHandles = partitionColumnHandles;
+        this.shouldSkipMetastoreForPartition = shouldSkipMetastoreForPartition;
+        this.fileSystemView = FileSystemViewManager.createInMemoryFileSystemView(engineContext, metaClient, metadataConfig);
+        this.partitionKeysFilter = MetastoreUtil.computePartitionKeyFilter(partitionColumnHandles, tableHandle.getPartitionPredicates());
+        this.partitionColumns = hiveTable.getPartitionColumns();
+    }
+
+    @Override
+    public List<HudiPartitionInfo> getPartitionsToScan()
+    {
+        HoodieTimer timer = new HoodieTimer().startTimer();
+
+        List<HudiPartitionInfo> allPartitionInfoList = null;
+
+        if (shouldSkipMetastoreForPartition) {
+            try {
+                // Use relative partition path and other context to construct
+                // InternalHudiPartitionInfo instances
+                PartitionValueExtractor partitionValueExtractor = partitionColumns.isEmpty()
+                        ? new NonPartitionedExtractor()
+                        : inferPartitionValueExtractorWithHiveMetastore();
+                List<String> relativePartitionPathList = partitionColumns.isEmpty()
+                        ? Collections.singletonList("")
+                        : TimelineUtils.getPartitionsWritten(metaClient.getActiveTimeline());
+                allPartitionInfoList = relativePartitionPathList.stream()
+                        .map(relativePartitionPath ->
+                                buildHudiPartitionInfo(
+                                        shouldSkipMetastoreForPartition,
+                                        Option.of(relativePartitionPath),
+                                        Option.empty(),
+                                        Option.of(partitionValueExtractor),
+                                        partitionColumns,
+                                        partitionColumnHandles,
+                                        tableHandle.getPartitionPredicates(),
+                                        hiveTable,
+                                        hiveMetastore))
+                        .collect(Collectors.toList());
+            }
+            catch (HoodieIOException e) {
+                log.warn("Cannot skip Hive Metastore for scanning partitions. Falling back to using Hive Metastore.");
```

Comment:
Is this actionable in some way? If so, include the table name and other information that an administrator might need.